### PR TITLE
feat(apm): added Sentry to the NLU server

### PR DESCRIPTION
**Pull request by michaelmass**

### CODE REVIEW

File: packages/nlu/src/nlu-server/app.ts [hunk 1]
```diff
@@ -52,6 +55,19 @@ export const createApp = async (
...
   app.use(cors())
 
+  if (options.sentryDSN) {
+    Sentry.init({ 
+      dsn: options.sentryDSN,
+      integrations: [
+        new Sentry.Integrations.Http({ tracing: true }),
+        new Tracing.Integrations.Express({ app })
+      ],
+      sampleRate: 1.0,
```
Comment (spg) [inline packages/nlu/src/nlu-server/app.ts:65]: Configurable sample rate would be better

Comment (spg) [reply packages/nlu/src/nlu-server/app.ts:65]: Also, missing: `app.use(Sentry.Handlers.tracingHandler());`

File: packages/nlu/src/nlu-server/app.ts [hunk 2]
```diff
@@ -52,6 +55,19 @@ export const createApp = async (
   // This must be first, otherwise the /info endpoint can't be called when token is used
   app.use(cors())
 
+  if (options.sentryDSN) {
+    Sentry.init({
```
Comment (spg) [inline packages/nlu/src/nlu-server/app.ts:59]: We should probably introduce a `SENTRY_ENVIRONMENT` env var that will be used for setting the Sentry Environment (https://docs.sentry.io/platforms/node/guides/express/configuration/environments/)

`SENTRY_ENVIRONMENT` could be `staging` or `prod`, for example.

File: packages/nlu/src/nlu-server/app.ts [hunk 3]
```diff
@@ -52,6 +55,19 @@ export const createApp = async (
   // This must be first, otherwise the /info endpoint can't be called when token is used
   app.use(cors())
 
+  if (options.sentryDSN) {
+    Sentry.init({ 
+      dsn: options.sentryDSN,
```
Comment (spg) [inline packages/nlu/src/nlu-server/app.ts:60]: Maybe we also want to set the `release` config? (https://docs.sentry.io/platforms/node/guides/express/configuration/options/#release)

**Review by spg — APPROVED** [review]

File: packages/nlu/src/index.ts [hunk 4]
```diff
@@ -62,6 +62,17 @@ yargs
         description: 'Authentification token for your language server',
         type: 'string'
       },
+      sentryEnabled: {
+        description:
+          'When enabled, Sentry is added to the express server allowing the use of the environment variables SENTRY_DSN, SENTRY_ENVIRONMENT, SENTRY_RELEASE',
+        default: process.env.SENTRY_ENABLED === 'true',
```
Comment (EFF) [inline packages/nlu/src/index.ts:68]: anywhere else we use the. yn package to parse bool env vars, takes 1,0, y,n yes/no, true,false t,f & so on

File: packages/nlu/src/index.ts [hunk 5]
```diff
@@ -62,6 +62,17 @@ yargs
         description: 'Authentification token for your language server',
         type: 'string'
       },
+      sentryEnabled: {
```
Comment (EFF) [inline packages/nlu/src/index.ts:65]: Can we rename those vars? Naming is tightly coupled with the vendor's name. If we end up changing APM provider, we'll need to break the interface which doesn't make sens

**Review by EFF — DISMISSED** [review]
minor changes, then LGTM

**Review by franklevasseur — APPROVED** [review]
:+1:

**Pull request merged**